packages/ember-data/lib/adapters/rest-adapter.js

```diff
@@ -774,11 +779,11 @@ export default Adapter.extend(BuildURLMixin, {
     @param {Object} options
     @return {Promise} promise
   */
-  ajax: function(url, type, options) {
+  ajax: function(request) {
```

Comment:
I've put everything behind a feature flag, so the above comment is not relevant anymore: the existing `ajax` function isn't touched.